www/inappbrowser.js

```diff
@@ -17,109 +17,136 @@
-            'message': channel.create('message')
-        };
-    }
+(function() {
+  // special patch to correctly work on Ripple emulator (CB-9760)
+  if (window.parent && !!window.parent.ripple) {
+    // https://gist.github.com/triceam/4658021
+    module.exports = window.open.bind(window); // fallback to default window.open behaviour
+    return;
+  }
 
-    InAppBrowser.prototype = {
-        _eventHandler: function (event) {
-            if (event && (event.type in this.channels)) {
-                if (event.type === 'beforeload') {
-                    this.channels[event.type].fire(event, this._loadAfterBeforeload);
-                } else {
-                    this.channels[event.type].fire(event);
-                }
-            }
-        },
-        _loadAfterBeforeload: function (strUrl) {
-            strUrl = urlutil.makeAbsolute(strUrl);
-            exec(null, null, 'InAppBrowser', 'loadAfterBeforeload', [strUrl]);
-        },
-        close: function (eventname) {
-            exec(null, null, 'InAppBrowser', 'close', []);
-        },
-        show: function (eventname) {
-            exec(null, null, 'InAppBrowser', 'show', []);
-        },
-        hide: function (eventname) {
-            exec(null, null, 'InAppBrowser', 'hide', []);
-        },
-        addEventListener: function (eventname, f) {
-            if (eventname in this.channels) {
-                this.channels[eventname].subscribe(f);
-            }
-        },
-        removeEventListener: function (eventname, f) {
-            if (eventname in this.channels) {
-                this.channels[eventname].unsubscribe(f);
-            }
-        },
+  var exec = require("cordova/exec");
+  var channel = require("cordova/channel");
+  var modulemapper = require("cordova/modulemapper");
+  var urlutil = require("cordova/urlutil");
 
-        executeScript: function (injectDetails, cb) {
-            if (injectDetails.code) {
-                exec(cb, null, 'InAppBrowser', 'injectScriptCode', [injectDetails.code, !!cb]);
-            } else if (injectDetails.file) {
-                exec(cb, null, 'InAppBrowser', 'injectScriptFile', [injectDetails.file, !!cb]);
-            } else {
-                throw new Error('executeScript requires exactly one of code or file to be specified');
-            }
-        },
-
-        insertCSS: function (injectDetails, cb) {
-            if (injectDetails.code) {
-                exec(cb, null, 'InAppBrowser', 'injectStyleCode', [injectDetails.code, !!cb]);
-            } else if (injectDetails.file) {
-                exec(cb, null, 'InAppBrowser', 'injectStyleFile', [injectDetails.file, !!cb]);
-            } else {
-                throw new Error('insertCSS requires exactly one of code or file to be specified');
-            }
-        }
+  function InAppBrowser() {
+    this.channels = {
+      beforeload: channel.create("beforeload"),
+      ondownload: channel.create("ondownload"),
```

Comment:
Despite all the whitespace and quote changes, this is the only real change, right? Adding `ondownload`? Is that the start of a download, or the finish of a download? I know in Electron, there is start, end, progress, and error.

Reply:
That really depends on the kind of response I send from the native side.